ProvinceMapper/Source/LinkMapper/LinkMapping.cpp

```diff
@@ -0,0 +1,102 @@
+#include "LinkMapping.h"
+#include "ParserHelpers.h"
+#include "Log.h"
+#include "../Provinces/Province.h"
+#include "../Definitions/Definitions.h"
+
+LinkMapping::LinkMapping(std::istream& theStream, const Definitions& sourceDefs, const Definitions& targetDefs)
+{
+	registerKeys(sourceDefs, targetDefs);
+	parseStream(theStream);
+	clearRegisteredKeywords();
+}
+
+void LinkMapping::registerKeys(const Definitions& sourceDefs, const Definitions& targetDefs)
+{
+	registerKeyword("ck3", [this, sourceDefs](const std::string& unused, std::istream& theStream) {
+		const auto id = commonItems::singleInt(theStream).getInt();
+		const auto& provinces = sourceDefs.getProvinces();
+		const auto& provItr = provinces.find(id);
+		if (provItr != provinces.end())
+			sources.emplace_back(provItr->second);
+	});
+	registerKeyword("eu4", [this, targetDefs](const std::string& unused, std::istream& theStream) {
+		const auto id = commonItems::singleInt(theStream).getInt();
+		const auto& provinces = targetDefs.getProvinces();
+		const auto& provItr = provinces.find(id);
+		if (provItr != provinces.end())
+			targets.emplace_back(provItr->second);
+	});
+	registerKeyword("comment", [this](const std::string& unused, std::istream& theStream) {
+		comment = commonItems::singleString(theStream).getString();
+	});
+	registerRegex(commonItems::catchallRegex, commonItems::ignoreItem);
+}
+
+std::ostream& operator<<(std::ostream& output, const LinkMapping& linkMapping)
+{
+	output << "\tlink = { ";
```

Comment:
It might be worth having functions handling different subsections.

Reply:
Within an output operator? It'd be really messy, messier than this I think. And it's not that unreadable. I'll put some comments.